### ISSUE TITLE
multiple string length issues in custom resources

### ISSUE THREAD

**Issue by ranman** [aws-quickstart/cdk-eks-blueprints#401]
### Describe the bug

There are multiple issues with string length in the custom resources for provisioning addons and the core cluster. There are similar issues in the tagging as well.

e.g. The code here: https://github.com/aws-quickstart/cdk-eks-blueprints/blame/6ba0edae08f13d99c6f596edc9e4fd6c72cb9d9d/lib/utils/vpc-utils.ts#L61-L81

Doesn't truncate tag keys to 128 characters which will cause the custom resource to fail and will cause stacks to fail if their name is too long. This happens most often when using the blueprints pipelines.

### Expected Behavior

Ideally, these kinds of failures would be detected at build time (e.g. synth) rather than deploying and failing because the custom resources aren't truncating or dealing with string length issues.

### Current Behavior

The stack will synth without error but will fail to deploy.

### Reproduction Steps

```typescript
import 'source-map-support/register';
import * as cdk from 'aws-cdk-lib';
import * as blueprints from '@aws-quickstart/eks-blueprints';

const app = new cdk.App();

const addOns: Array<blueprints.ClusterAddOn> = [
  new blueprints.addons.MetricsServerAddOn(),
  new blueprints.addons.ClusterAutoScalerAddOn(),
  new blueprints.addons.VpcCniAddOn(),
];

blueprints.EksBlueprint.builder()
    .addOns(...addOns)
    .build(app, 'eks-blueprint-with-an-extra-kind-of-very-long-name-that-is-definitely-not-going-to-work');
````

```
CREATE_FAILED        | AWS::EKS::Nodegroup                   | eks-blueprint-with...oing-to-work-ng-ng
Resource handler returned message: "nodeGroupName can't be longer than 63 characters! (Service: Eks, Status Code: 400, Request ID: c31c5853-f46c-4e32-9f23-8
```

### CDK CLI Version

2.25.0 (build ae1cb4b)

### EKS Blueprints Version

1.0.1

### Node.js Version

v18.2.0

### Environment details (OS name and version, etc.)

mac

### Other information

Other failure stacktraces:

```1 validation error detected: Value 'k8s.io/cluster-autoscaler/twitter-for-pets-eks-blueprint-pipeline/twitter-for-pets-blueprints-prod/twitter-for-pets-blueprints-prod-blueprint/clusterName' at 'tags.2.member.key' failed to satisfy constraint: Member must have length less than or equal to 128 (Service: AmazonIdentityManagement; Status Code: 400; Error Code: ValidationError; Request ID: 83441c5c-d23d-48b3-ad8f-597f7425df51; Proxy: null)```

**Comment by shapirov103** [comment]
@ranman this was fixed (based on our testing) in #424 and release with 1.1.0. If you get a chance, could you verify from your end and close if everything works as expected.

**Comment by shapirov103** [comment]
If you observe anything that can be fixed, please let us know. Fixed with #424
